### ISSUE TITLE
Defined child nodes of document should not be deletable 

### ISSUE THREAD

**Issue by jonnitto** [neos/neos-ui#2620]
### Description

When a node is defined as child nodes, the editor should not be able to delete it. Since version 5.1 it is possible. 

### Steps to Reproduce

Add a child node to the homepage

```yaml
'Foo.Bar:Document.Home':
  childNodes:
    imprint:
      type: 'Foo.Bar:Document.Page'
```

Run `./flow node:repair` to create the node

#### Expected behavior

The editor should not be able to delete the node (behavior in 5.0)
![Neos 5.0](https://user-images.githubusercontent.com/4510166/70853561-a2f07880-1eaf-11ea-9782-f9e887ccb565.png)

#### Actual behavior

The editor is able to delete the node (behavior in 5.1)
![Neos 5.1](https://user-images.githubusercontent.com/4510166/70853576-cd423600-1eaf-11ea-9a37-54c3263dd40e.png)

### Affected Versions

Neos: 5.1.0

UI: 5.1.0

**Comment by dimaip** [comment]
Was able to reproduce, will fix

**Comment by skurfuerst** [comment]
solved in 5.1.1.
